### PR TITLE
sublime-music: 0.11.10 -> 0.11.11

### PR DESCRIPTION
**Pull request by sumnerevans**
This upgrade includes a switch from setuptools to poetry for the build. It also converts to using the test suite provided by the Sublime Music upstream project rather than just the import check.

###### Motivation for this change

https://gitlab.com/sublime-music/sublime-music/-/releases/v0.11.11

###### Things done

```
$ nix-shell -p nixpkgs-review --run "nixpkgs-review rev ad6389e618987a29371cb5e506208400248868b9"
...
[nix-shell ...] $ sublime-music
```

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/sublime-music/default.nix [hunk 1]
```diff
@@ -19,7 +27,7 @@ python3Packages.buildPythonApplication rec {
 
   nativeBuildInputs = [
     gobject-introspection
-    python3Packages.setuptools
+    python3Packages.poetry
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/sublime-music/default.nix:30]: Can we use poetry-core?

File: pkgs/applications/audio/sublime-music/default.nix [hunk 2]
```diff
@@ -53,9 +61,12 @@ python3Packages.buildPythonApplication rec {
   # https://github.com/NixOS/nixpkgs/issues/56943
   strictDeps = false;
 
-  # no tests
-  doCheck = false;
-  pythonImportsCheck = [ "sublime_music" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/sublime-music/default.nix:58]: Why did you remove the importsCheck?

Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:58]: I am now using the upstream unit tests directly, rather than relying on the imports check. https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck says that it's "highly preferred" to use unit tests over the `pythonImportsCheck`.

If you want I can add it back, but it's redundant considering that the unit tests import the module and much more.

File: pkgs/applications/audio/sublime-music/default.nix (hunk 1 above)
Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:30]: I get this error:
```
ModuleNotFoundError: No module named 'poetry.masonry'
```
when I use `poetry-core` instead of `poetry`.

**Comment by sumnerevans** [comment]
@albakham would you be able to review this as well?

File: pkgs/applications/audio/sublime-music/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/sublime-music/default.nix:30]: You can fetchpatch https://gitlab.com/sublime-music/sublime-music/-/commit/9b0af19dbdfdcc5a0fa23e73bb34c7135a8c2855.patch from https://gitlab.com/sublime-music/sublime-music/-/merge_requests/60

File: pkgs/applications/audio/sublime-music/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/sublime-music/default.nix:58]: checkInput can contain extra packages which can hide import errors. It doesn't hurt to add a pythonImportsCheck and adds some safety to not miss depedencies.
